backend/src/apiserver/resource/resource_manager.go

```diff
@@ -415,6 +416,14 @@ func (r *ResourceManager) CreateRun(apiRun *api.Run) (*model.RunDetail, error) {
 		return nil, err
 	}
 
+	_, err = validate.ValidateWorkflow(nil, nil, workflow.Workflow, validate.ValidateOpts{
+		Lint:                       false,
+		IgnoreEntrypoint:           false,
+		WorkflowTemplateValidation: false, // not used by kubeflow
+	})
+	if err != nil {
+		return nil, util.NewInternalServerError(err, "Failed to validate workflow for (%+v)", workflow.Workflow.Name)
+	}
```

Comment:
Hi @NikeNano, I am just thinking about the high-level value this extra validation brings:
* we'll find out about invalid workflows faster

However, this also creates stronger coupling to argo -- if there are problems with validation, we'll have to upgrade/downgrade argo go module more carefully.

Therefore, I think more value is needed to justify the cost. and I think we should introduce this validation in more cases:
* when uploading a pipeline
* when creating a recurring run

In these cases, if we follow the normal execution, only when the pipeline runs will we know that argo workflow is in fact not valid, so with this new feature -- we'll be able to verify validity right  away, that's amazing!

Reply:
Agree make sens to me, when uploading a pipeline might actually be the occasions that is the most valuable to the user.

Reply:
Thanks for pointing it out, I might have missed it

Reply:
Do you agree or see something we should change with the current implementation @Bobgy

Reply:
I think it make sens to keep all the validation in one place for now, we do some other validation within `GetParameters` currently. https://github.com/kubeflow/pipelines/blob/e5af2a9e36bc4e56e7797fa3a8fae9e19d5773d4/backend/src/common/util/template_util.go#L48-L68